controllers/addons_controller.go

```diff
@@ -90,23 +97,28 @@ func NewReconciler(config *rest.Config, client client.Client, logger logr.Logger
 		Log:    logger.WithName("reconciler"),
 		Scheme: scheme,
 	}
-	reconciler.k8client = reconciler.getK8sClient()
-	reconciler.Context = context.Background()
 	var err error
+	reconciler.k8client, err = reconciler.getK8sClient()
+	if err != nil {
+		return nil, err
+	}
+	reconciler.Context = context.Background()
 	reconciler.Applier, err = apply.NewApplier(client, logger.WithName("applier"), scheme)
+	if err != nil {
+		return nil, err
```

Comment:
Should we wrap this error before returning it?

Reply:
done, we should use errors.Wrap and WithMessage etc more widely per https://itnext.io/golang-error-handling-best-practice-a36f47b0b94c